quill-core/src/test/scala/io/getquill/dsl/MetaDslSpec.scala

```diff
@@ -73,10 +73,24 @@ class MetaDslSpec extends Spec {
       "> 22 fields" in {
         val meta = materializeQueryMeta[MoreThan22]
         meta.expand.toString mustEqual "(q) => q.map(x => (x.v0, x.v1, x.v2, x.v3, x.v4, x.v5, x.v6, x.v7, x.v8, x.v9, x.x0, x.x1, x.x2, x.x3, x.x4, x.x5, x.x6, x.x7, x.x8, x.x9, x.y0, x.y1, x.y2, x.y3, x.y4, x.y5, x.y6, x.y7, x.y8, x.y9))"
-        meta.extract(Row((0 until 30): _*)) mustEqual
+        meta.extract(Row(0 until 30: _*)) mustEqual
           MoreThan22(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29)
       }
     }
+    "private val" - {
+      "only" in {
+        case class Entity(private val a: Int)
+        val meta = materializeQueryMeta[Entity]
+        meta.expand.toString mustEqual "(q) => q.map(x => ())"
+        meta.extract(Row(1)) mustEqual Entity(1)
+      }
+      "mixed" in {
+        case class Entity(a: Int, private val b: Int)
+        val meta = materializeQueryMeta[Entity]
+        meta.expand.toString mustEqual "(q) => q.map(x => x.a)"
+        meta.extract(Row(1, 2)) mustEqual Entity(1, 2)
```

Comment:
Hm.. It seems I understood what you mean. I added failing test case. Actually the above `meta` should expand to `(q) => q.map(x => (x.a, x.b))` too but it can not because if it does not filter out private vals from `elements` [here](https://github.com/mxl/quill/blob/7316aac1565a463de55101b77ec7167fee757a58/quill-core/src/main/scala/io/getquill/dsl/MetaDslMacro.scala#L70) then compilation fails.